sdk/python/kfp/cli/cli_test.py

```diff
@@ -279,6 +279,85 @@ def test_compile_with_caching_disabled_env_var(self):
             caching_options = task['cachingOptions']
             self.assertEqual(caching_options, {})
 
+    def test_compile_with_kubernetes_manifest_format(self):
+        temp_pipeline = tempfile.NamedTemporaryFile(suffix='.py', delete=False)
```

Comment:
Can you use this as a context manager and set `delete=True` to ensure clean up after the test?